src/components/Modal/Modal.module.scss

```diff
@@ -13,33 +20,34 @@
         --rui-local-max-width: calc(100% - (2 * var(--rui-local-outer-spacing)));
         --rui-local-max-height: calc(100% - (2 * var(--rui-local-outer-spacing)));
 
-        position: fixed;
-        left: 50%;
-        z-index: settings.$z-index;
-        display: flex;
         flex-direction: column;
         max-width: var(--rui-local-max-width);
         max-height: var(--rui-local-max-height);
+        padding: 0;
         overflow-y: auto;
         overscroll-behavior: contain;
+        border-width: 0;
         border-radius: settings.$border-radius;
         background: theme.$background;
         box-shadow: theme.$box-shadow;
-        transform: translateX(-50%);
 
         @include breakpoint.up(sm) {
             --rui-local-outer-spacing: #{theme.$outer-spacing-sm};
         }
     }
 
-    .backdrop {
-        position: fixed;
-        top: 0;
-        left: 0;
-        z-index: settings.$backdrop-z-index;
-        width: 100vw;
-        height: 100vh;
+    .root:has(.root)::backdrop {
```

Comment:
⚠️ `<dialog>`s rather should not be nested like that. It is OK to stack `<dialog>`s visually, but in practice, it should mean they are siblings in the DOM. While the HTML spec does not explicitly discourage from nesting one `<dialog>` into another in the DOM (and while it seems to "work" somehow), we are not quite sure of the consequences.

OK:

```html
<dialog></dialog>
<dialog></dialog>
```
⚠️ Not OK:

```html
<dialog>
  <dialog></dialog>
</dialog>
```

https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web-react/src/components/Modal#stacking-modals